### PR TITLE
REFACTOR: kwargs

### PR DESCRIPTION
**Pull request by gmalinve**
refactoring kwargs maxwellcircuit.py, CableModeling.py, report_templates.py, general_methods.py

### CODE REVIEW

File: pyaedt/modules/CableModeling.py [hunk 1]
```diff
@@ -109,6 +110,7 @@ def __init__(self, app, json_file_name=None, working_dir=None):
         if json_file_name:
             self._init_from_json(json_file_name)
 
+    @pyaedt_function_handler()
```
Comment (MaxJPRey) [inline pyaedt/modules/CableModeling.py:113]: Why do we need the handler function decorator if no modifications are made to the methods' signatures? No args modified it seems.

Comment (gmalinve) [reply pyaedt/modules/CableModeling.py:113]: @MaxJPRey this class would need a refactor so I wasn't sure to touch it or not. I'd like to get rid of all the try/except blocks and use the decorator to catch any exception if anything fails but I'd like to that in a different PR.

File: _unittest/test_02_2D_modeler.py [hunk 2]
```diff
@@ -135,12 +135,12 @@ def test_07_create_ellipse(self):
         assert ellipse1.solve_inside
         assert ellipse1.model
         assert ellipse1.material_name == "vacuum"
-        assert isclose(ellipse2.faces[0].area, math.pi * 4.0 * 4.0 * 3, rel_tol=0.1)
+        assert isclose(ellipse2.faces[0].area, math.pi * 4.0 * 4.0 * 3, relative_tolerance=0.1)
```
Comment (maxcapodi78) [inline _unittest/test_02_2D_modeler.py:138]: maybe just tolerance?

File: examples/03-Maxwell/Maxwell_Transformer_Coreloss.py [hunk 3]
```diff
@@ -36,22 +36,22 @@
 freq_curve_csv_700kHz = downloads.download_file("core_loss_transformer", "mf3_700kHz.csv", temp_folder)
 freq_curve_csv_1MHz = downloads.download_file("core_loss_transformer", "mf3_1MHz.csv", temp_folder)
 
-data = read_csv_pandas(filename=freq_curve_csv_25kHz)
+data = read_csv_pandas(file_name=freq_curve_csv_25kHz)
```
Comment (maxcapodi78) [inline examples/03-Maxwell/Maxwell_Transformer_Coreloss.py:39]: this should be input_file

File: pyaedt/generic/general_methods.py [hunk 4]
```diff
@@ -289,9 +289,9 @@ def check_numeric_equivalence(a, b, relative_tolerance=1e-7):
     return True if reldiff < relative_tolerance else False
 
 
-@pyaedt_function_handler()
-def _check_path(path_to_check):
-    return path_to_check.replace("\\", "/") if path_to_check[0] != "\\" else path_to_check
+@pyaedt_function_handler(path_to_check="path")
```
Comment (maxcapodi78) [inline pyaedt/generic/general_methods.py:292]: I don't think we need to decorate with kwargs internal methods

**Review by maxcapodi78 — CHANGES_REQUESTED** [review]

File: pyaedt/modules/CableModeling.py (hunk 1 above)
Comment (MaxJPRey) [reply pyaedt/modules/CableModeling.py:113]: OK, got it. Thanks @gmalinve.

File: pyaedt/application/analysis_hf.py [hunk 5]
```diff
@@ -118,23 +122,23 @@ def get_all_insertion_loss_list(
...
-
-        if reclist is None:
-            reclist = [i for i in list(self._app.excitations)]
-        if tx_prefix:
-            trlist = [i for i in trlist if i.startswith(tx_prefix)]
-        if rx_prefix:
-            reclist = [i for i in reclist if i.startswith(rx_prefix)]
+        if drivers is None:
+            drivers = [i for i in list(self._app.excitations)]
+
```
Comment (SMoraisAnsys) [inline pyaedt/application/analysis_hf.py:127]:
```suggestion
```

File: pyaedt/generic/general_methods.py (hunk 4 above)
Comment (SMoraisAnsys) [reply pyaedt/generic/general_methods.py:292]: Agreed, do we have a way to know if some users use any of them ?

**Review by SMoraisAnsys — APPROVED** [review]
Great refactoring ! I left minor comments

File: pyaedt/generic/general_methods.py (hunk 4 above)
Comment (maxcapodi78) [reply pyaedt/generic/general_methods.py:292]: if they use it is at their risk. The intent of a private method should be to not be used by the end user

Comment (Samuelopez-ansys) [reply pyaedt/generic/general_methods.py:292]: I think this discussion is related to https://github.com/ansys/pyaedt/issues/4666

File: _unittest/test_02_2D_modeler.py (hunk 2 above)
Comment (gmalinve) [reply _unittest/test_02_2D_modeler.py:138]: @maxcapodi78 the fact is that the method isclose accepts two different tolerances: relative and absolute.

File: _unittest/test_12_PostProcessing.py [hunk 6]
```diff
@@ -373,18 +373,18 @@ def test_18_diff_plot(self, diff_test):
...
-        assert data1.plot(formula="phasedeg", snapshot_path=os.path.join(self.local_scratch.path, "temp8.jpg"))
-        assert data1.plot(formula="phaserad", snapshot_path=os.path.join(self.local_scratch.path, "temp9.jpg"))
+        assert data1.plot("S(Diff1, Diff1)")
+        assert data1.plot()
+        assert data1.plot()
+        assert data1.plot()
+        assert data1.plot()
+        assert data1.plot()
+        assert data1.plot()
+        assert data1.plot()
```
[the remark this answers is not in the document]
Comment (gmalinve) [reply _unittest/test_12_PostProcessing.py:387]: @SMoraisAnsys yes agree, thanks for catching that. I didn't see it!

File: examples/02-HFSS/HFSS_Choke.py [hunk 7]
```diff
@@ -12,7 +12,8 @@
 import os
 import pyaedt
 
-project_name = pyaedt.generate_unique_project_name(rootname=r"C:\Data\Support\Test", folder_name="choke", project_name="choke")
+project_name = pyaedt.generate_unique_project_name(root_name=r"C:\Data\Support\Test", folder_name="choke",
```
[the remark this answers is not in the document]
Comment (gmalinve) [reply examples/02-HFSS/HFSS_Choke.py:15]: @SMoraisAnsys agree, I'll remove it!

**Review by SMoraisAnsys — APPROVED** [review]

**Review by SMoraisAnsys — APPROVED** [review]
LGTM, a lot of work for sure ! Thanks again with this refactoring effort !

**Pull request merged**